File: nesoy.kwon/src/main/java/homework/chapter_6/Member.java
package homework.chapter_6;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;

import java.util.List;

@Entity
public class Member {

	@Id
	@Column(name = "MEMBER_ID")
	private String id;

	@OneToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "LOCKER_ID")
	private Locker locker;

	private String username;

	@OneToMany(mappedBy = "member")
	private List<MemberProduct> memberProducts;

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public Locker getLocker() {
		return locker;
	}

	public void setLocker(Locker locker) {
		this.locker = locker;
	}

	@Override
	public String toString() {
		return "Member{" +
				"id='" + id + '\'' +
				", username='" + username + '\'' +
				'}';
	}
}

File: hyun.k/hyun.k/src/main/kotlin/chapter6/homework1/Member.kt
package chapter6.homework1

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity(name = "member")
data class Member(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null, // val로 만들면 위임이 불가능해짐.

    @Column(length = 50, nullable = false)
    val name: String,
)
